src/sagemaker/deserializers.py

```diff
@@ -13,312 +13,101 @@
 from __future__ import absolute_import
 
-import csv
 
-import abc
-import codecs
-import io
-import json
-
-import numpy as np
-from six import with_metaclass
-
-from sagemaker.utils import DeferredError
-
-try:
-    import pandas
-except ImportError as e:
-    pandas = DeferredError(e)
-
-
-class BaseDeserializer(abc.ABC):
-    """Abstract base class for creation of new deserializers.
-
-    Provides a skeleton for customization requiring the overriding of the method
-    deserialize and the class attribute ACCEPT.
+from typing import List, Optional
+
+# base_deserializers was renamed from deserializers, so this import
+# is for backwards compatibility.
+from sagemaker.base_deserializers import *  # noqa: F403, F401 # pylint: disable=W0614,W0401
+
+from sagemaker.base_deserializers import BaseDeserializer
+
+from sagemaker.jumpstart import artifacts, utils as jumpstart_utils
+
+
+def retrieve(
+    region: Optional[str] = None,
+    model_id: Optional[str] = None,
+    model_version: Optional[str] = None,
+    tolerate_vulnerable_model: bool = False,
+    tolerate_deprecated_model: bool = False,
+) -> List[BaseDeserializer]:
+    """Retrieves the supported deserializers for the model matching the given arguments.
+
+    Args:
+        region (str): The AWS Region for which to retrieve the supported deserializers.
+            Defaults to ``None``.
+        model_id (str): The model ID of the model for which to
+            retrieve the supported deserializers. (Default: None).
+        model_version (str): The version of the model for which to retrieve the
+            supported deserializers. (Default: None).
+        tolerate_vulnerable_model (bool): True if vulnerable versions of model
+            specifications should be tolerated (exception not raised). If False, raises an
+            exception if the script used by this version of the model has dependencies with known
+            security vulnerabilities. (Default: False).
+        tolerate_deprecated_model (bool): True if deprecated models should be tolerated
+            (exception not raised). False if these models should raise an exception.
+            (Default: False).
+    Returns:
+        List[BaseDeserializer]: The supported deserializers to use for the model.
+
+    Raises:
+        ValueError: If the combination of arguments specified is not supported.
     """
 
-    @abc.abstractmethod
-    def deserialize(self, stream, content_type):
-        """Deserialize data received from an inference endpoint.
-
-        Args:
-            stream (botocore.response.StreamingBody): Data to be deserialized.
-            content_type (str): The MIME type of the data.
-
-        Returns:
-            object: The data deserialized into an object.
-        """
-
-    @property
-    @abc.abstractmethod
-    def ACCEPT(self):
-        """The content types that are expected from the inference endpoint."""
-
-
-class SimpleBaseDeserializer(with_metaclass(abc.ABCMeta, BaseDeserializer)):
```

Comment:
Now realizing that what I reviewed above may have been already in the SDK. Some of the proposed changes I made there may no longer apply then.